apps/pay/components/Layouts/UsernameLayout.tsx

```diff
@@ -0,0 +1,189 @@
+"use client"
+
+import Link from "next/link"
+import copy from "copy-to-clipboard"
+import { useRouter, useSearchParams } from "next/navigation"
+import React, { useEffect, useState } from "react"
+import { Image, OverlayTrigger, Tooltip } from "react-bootstrap"
+
+import { getClientSidePayDomain } from "../../config/config"
+
+import styles from "./app-layout.module.css"
+
+type Props = {
+  children: React.ReactPortal | React.ReactNode
+  username: string
+}
+
+const UsernameLayoutContainer = ({ children, username }: Props) => {
+  const router = useRouter()
+  const searchParams = useSearchParams()
+  const memo = searchParams.get("memo")
+  const [openSideBar, setOpenSideBar] = React.useState<boolean>(false)
+  const [copied, setCopied] = React.useState<boolean>(false)
+  const [isClient, setIsClient] = useState(false)
```

Comment:
I don't really understand what `isClient` is for. could we drop it? seems it starts with false and is immediately set to true 3 lines above, without more logic to that. seems like dead code to me.

Reply:
this is because of this part: https://github.com/GaloyMoney/galoy/pull/3992/files#diff-f0ad19133e981f4e00ed785d1af692726d2fef1a9fc7a2726721e262955cc9abR103-R111


It is giving a React minified error (external content does not match server-rendered HTML), so here, we want to only show this if the client is ready. This is a quick fix for now.

Reply:
add a comment in the code then to explain why we have this. hacky workaround are better documented if they can be confusing (it is confusing to me still, but that's fine I don't know the intricacies of next)